toqito/matrices/fourier.py

```diff
@@ -48,13 +46,19 @@ def fourier(dim: int) -> np.ndarray:
     .. bibliography::
         :filter: docname in docnames
 
+    Parameters
+    ----------
+    dim : int
+        The size of the Fourier matrix.
 
-    :param dim: The size of the Fourier matrix.
-    :return: The Fourier matrix of dimension :code:`dim`.
-
+    Returns
+    -------
+    np.ndarray
+        The Fourier matrix of dimension :code:`dim`.
```

Comment:
Similar thing here, we don't want to specify the type of the object being returned, this is type-hinted already and makes for more difficult refactoring if specified in multiple places.

Reply:
Not just that, the return doctoring that was there before did not need to be changed.

Additionally the section headings should be delimited with = not -. Again, refer to any other doctoring in a function within toqito for reference

Reply:
Almost. Instead of the 

```
    References
    ==========
    .. bibliography::
        :filter: docname in docnames

    Parameters
    ==========
    dim : int
        The size of the Fourier matrix.

    Returns
    =======
        The Fourier matrix of dimension :code:`dim`.
```

we would have instead (as it was before):
```
    References
    ==========
    .. bibliography::
        :filter: docname in docnames


    :param dim: The size of the Fourier matrix.
    :return: The Fourier matrix of dimension :code:`dim`.
```